modules/plugin/imagemosaic/src/main/java/org/geotools/gce/imagemosaic/GranuleDescriptor.java

```diff
@@ -962,362 +1026,270 @@ public GranuleLoadingResult loadRaster(
-        final ReferencedEnvelope bbox =
-                useFootprint
-                        ? new ReferencedEnvelope(
-                                granuleBBOX.intersection(inclusionGeometry.getEnvelopeInternal()),
-                                granuleBBOX.getCoordinateReferenceSystem())
-                        : granuleBBOX;
-        boolean doFiltering = false;
-        if (filterMe && useFootprint) {
-            doFiltering = Utils.areaIsDifferent(inclusionGeometry, baseGridToWorld, granuleBBOX);
-        }
+        final RequestParams requestParams =
+                prepareRequest(
+                        imageReadParameters, index, cropBBox, mosaicWorldToGrid, request, hints);
 
-        // intersection of this tile bound with the current crop bbox
-        final ReferencedEnvelope intersection =
-                new ReferencedEnvelope(
-                        bbox.intersection(cropBBox), cropBBox.getCoordinateReferenceSystem());
-        if (intersection.isEmpty()) {
-            if (LOGGER.isLoggable(java.util.logging.Level.FINE)) {
-                LOGGER.fine(
-                        new StringBuilder("Got empty intersection for granule ")
-                                .append(this.toString())
-                                .append(" with request ")
-                                .append(request.toString())
-                                .append(" Resulting in no granule loaded: Empty result")
-                                .toString());
-            }
+        if (!boundsOverlap(requestParams)) {
             return null;
         }
 
-        // check if the requested bbox intersects or overlaps the requested area
-        if (useFootprint
-                && inclusionGeometry != null
-                && !JTS.toGeometry(cropBBox).intersects(inclusionGeometry)) {
-            if (LOGGER.isLoggable(java.util.logging.Level.FINE)) {
-                LOGGER.fine(
-                        new StringBuilder("Got empty intersection for granule ")
-                                .append(this.toString())
-                                .append(" with request ")
-                                .append(request.toString())
-                                .append(" Resulting in no granule loaded: Empty result")
-                                .toString());
+        GranuleLoadingResult result = null;
+
+        // eventually gets closed in finally block, if possible (not deferred loading)
+        final ReaderState readerState = prepareReader(requestParams);
+        try {
+            if (readerState.proceed) {
+                final RasterState rasterState = loadAndPrepareRaster(requestParams, readerState);
+                if (rasterState.proceed) {
+                    result = createGranuleLoadingResult(requestParams, readerState, rasterState);
+                }
             }
+        } catch (RuntimeException e) {
+            warnLoadRaster(e, request);
```

Comment:
The original code was also catching TransformException. Is that being taken into account?